lib/_stream_readable.js

```diff
@@ -68,11 +68,19 @@ function ReadableState(options, stream) {
   // the point at which it stops calling _read() to fill the buffer
   // Note: 0 is a valid value, means "don't call _read preemptively ever"
   var hwm = options.highWaterMark;
-  var defaultHwm = this.objectMode ? 16 : 16 * 1024;
-  this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
+  if (hwm != null) {
+    if (typeof hwm !== 'number') {
+      throw new TypeError('highWaterMark must be a number');
```

Comment:
I think we're standardizing on putting the offending data's name in double quotes. So `"highWaterMark"` in this case.